test/flow/codeGeneration.js

```diff
@@ -0,0 +1,319 @@
+import { expect } from 'chai';
+
+import { stripIndent } from 'common-tags';
+
+import {
+  parse,
+  isType,
+  GraphQLID,
+  GraphQLString,
+  GraphQLInt,
+  GraphQLList,
+  GraphQLNonNull
+} from 'graphql';
+
+import {
+  generateSource
+} from '../../src/flow/codeGeneration';
+
+import { loadSchema } from '../../src/loading';
+const schema = loadSchema(require.resolve('../starwars/schema.json'));
+
+import CodeGenerator from '../../src/utilities/CodeGenerator';
+
+import { compileToIR } from '../../src/compilation';
+
+describe('Flow code generation', function() {
+  beforeEach(function() {
+    const context = {
+      schema: schema,
+      operations: {},
+      fragments: {},
+      typesUsed: {}
+    }
+
+    this.generator = new CodeGenerator(context);
+
+    this.compileFromSource = (source) => {
+      const document = parse(source);
+      const context = compileToIR(schema, document);
+      this.generator.context = context;
+      return context;
+    };
+
+    this.addFragment = (fragment) => {
+      this.generator.context.fragments[fragment.fragmentName] = fragment;
+    };
+  });
+
+  describe('#generateSource()', function() {
+    it(`should generate simple query operations`, function() {
+      const context = this.compileFromSource(`
+        query HeroName {
+          hero {
+            name
+          }
+        }
+      `);
+
+      const source = generateSource(context);
+
+      expect(source).to.include(stripIndent`
+        /* @flow */
+        //  This file was automatically generated and should not be edited.
+
+        export type HeroNameQuery = {
+          hero: ? {
```

Comment:
The way code generation works doesn't let me (yet?) have those (`?` & `{`) correctly concatened. However, it is compliant with what we would do in other places!